internal/metrics/metrics.go

```diff
@@ -16,16 +16,28 @@
 
 package metrics
 
-import "github.com/prometheus/client_golang/prometheus"
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
 
 var registry *prometheus.Registry
+var BatchPinCounter prometheus.Counter
+
+// MetricsBatchPin is the prometheus metric for total number of batch pins submitted
+var MetricsBatchPin = "ff_batchPin_total"
 
 // Registry returns FireFly's customized Prometheus registry
 func Registry() *prometheus.Registry {
 	if registry == nil {
+		BatchPinCounter = prometheus.NewCounter(prometheus.CounterOpts{
```

Comment:
Since there will be a lot of metrics eventually would almost suggest creating a private method like `initMetricCollectors` or something.

Reply:
yep, I'll add this for both like you suggested, it'll be good to set the precedent now.